src/Workspaces/VisualBasic/Portable/Recommendations/VisualBasicRecommendationService.vb

```diff
@@ -247,6 +247,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Recommendations
             Dim leftHandTypeInfo = context.SemanticModel.GetTypeInfo(leftExpression, cancellationToken)
             Dim leftHandSymbolInfo = context.SemanticModel.GetSymbolInfo(leftExpression, cancellationToken)
 
+            ' GitHub #9087: Try to speculatively bind a type as an expression for My namespace
+            If leftHandTypeInfo.Type IsNot Nothing AndAlso leftHandSymbolInfo.Symbol Is leftHandTypeInfo.Type Then
```

Comment:
@AlekseyTs says this is how this is supposed to work in https://github.com/dotnet/roslyn/issues/9087#issuecomment-250325470